### PR TITLE
http: store stats by status code

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
Adding an option to aggregate http metrics by status code rather than by status code class.

### Motivation
A better stats

### Describe how to test/QA your changes
1. Run the agent and enable the feature `enable_http_status_code_aggregation = true` under `service_monitoring_config`
2. run `curl http://httpbin.org/status/203`
3. run ` sudo curl --unix /opt/datadog-agent/run/sysprobe.sock http://unix/network_tracer/debug/http_monitoring | jq`
4. make sure you have the relevant entry with 203 status code
5. re-run the agent now with `enable_http_status_code_aggregation = false`
6. re-run setup 2-3
7. make sure you have the relevant entry with 200 status code

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by Kaderinho — APPROVED** [review]

**Review by hmahmood — COMMENTED** [review]
Can we get a mem/cpu comparison to master for this? Also, you may want to remove the skip-qa tag since you have qa instructions.

**Comment by guyarb** [comment]
> Can we get a mem/cpu comparison to master for this? Also, you may want to remove the skip-qa tag since you have qa instructions.

mem/cpu - ill add the comparison
skip-qa - I have automatic tests that covers the feature, I added the qa steps to allow the code reviewer to play with the feature while reviewing it

File: pkg/config/system_probe.go [hunk 1]
```diff
@@ -172,6 +172,7 @@ func InitSystemProbeConfig(cfg Config) {
 
 	cfg.BindEnvAndSetDefault(join(smNS, "enable_java_tls_support"), false)
 	cfg.BindEnvAndSetDefault(join(smNS, "java_agent_args"), defaultServiceMonitoringJavaAgentArgs)
+	cfg.BindEnvAndSetDefault(join(smNS, "enable_http_status_code_aggregation"), false)
```
Comment (brycekahle) [inline pkg/config/system_probe.go:175]: I'd suggest renaming this. IMHO enabling aggregation feels like it should result in less stats, not more.

Comment (guyarb) [reply pkg/config/system_probe.go:175]: How about `enable_http_status_code_collection`?

**Review by Yumasi — APPROVED** [review]

**Comment by guyarb** [comment]
Original version
image

With the change
image

**Review by derekwbrown — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

**Pull request merged**